pgjdbc/src/main/java/org/postgresql/core/ByteOptimizedUTF8Encoder.java

```diff
@@ -22,7 +22,7 @@
   public String decode(byte[] encodedString, int offset, int length) throws IOException {
     //for very short strings going straight to chars is up to 30% faster
     if (length <= 32) {
-      return charDecode(encodedString, offset, length);
+      return length > 0 ? charDecode(encodedString, offset, length) : "";
```

Comment:
This and the same change in CharOptimizedUTF8Encoder should be their own commit as they're unrelated improvements.